### ISSUE TITLE
Problem using catboost in an ensemble

### ISSUE THREAD

**Issue by mcaprio2** [mlr-org/mlr3#669]
I have been helping  student with some FTIR spectrophotometry analysis, trying to use several models to build an ensemble.  MLR3 seemed like a great way to do this and we decided to include a catboost model as it did well in initial tests.  Wen've run into a problem where the predictions from catboost seem to be problematic.  I've created a minimal example using the Indian Pines hyperspectral dataset (http://www.ehu.eus/ccwintco/index.php/Hyperspectral_Remote_Sensing_Scenes#Indian_Pines) and recreated the problem using a two model ensemble.  I believe the script is functional as it works substituting an xgboost model for the catboost model.  Utlimately the model fails at:
INFO  [13:14:26.204] [bbotk] Starting to optimize 6 parameter(s) with '<OptimizerGenSA>' and '<TerminatorEvals> [n_evals=10]'
INFO  [13:14:26.241] [bbotk] Evaluating 1 configuration(s)
INFO  [13:14:26.394] [mlr3]  Running benchmark with 1 resampling iterations
INFO  [13:14:26.827] [mlr3]  Applying learner 'svm_1.cat_1.union1.classif.ranger' on task 'CatBoostTask' (iter 1/1)
INFO  [13:17:35.099] [mlr3]  Applying learner 'classif.svm' on task 'CatBoostTask' (iter 2/3)
INFO  [13:19:09.092] [mlr3]  Applying learner 'classif.svm' on task 'CatBoostTask' (iter 1/3)
INFO  [13:20:43.884] [mlr3]  Applying learner 'classif.svm' on task 'CatBoostTask' (iter 3/3)
INFO  [13:35:52.753] [mlr3]  Applying learner 'classif.catboost' on task 'CatBoostTask' (iter 2/3)
Error in check_prediction_data.PredictionDataClassif(pdata) :
  Assertion on 'colnames(prob)' failed: Must have names.

I noticed there was a recent problem reported with resample and catboost.  We have also had similar problems with catboost and benchmarks.  I am uncertain if the root error is the same or if these are two different problems. I have no problem tuning catboost with TuningInstanceSingleCrit, but resampling, benchmarking and the ensemble just won't work. 

The script used was:
library(R.matlab)
library("mlr3")
library("mlr3viz")
library("mlr3learners")
library(mlr3extralearners)
library("mlr3pipelines")
library("paradox")
library("mlr3tuning")
library("ggplot2")
library("mlr3benchmark")
library("mlr3hyperband")
library("mlr3filters")
library("mlr3fselect")
library("praznik")

library(mlrintermbo)
library(bbotk)

future::plan("multisession")

Distribution<-readMat("./Indian_pines_corrected.mat")
DataLabels<-readMat("./Indian_pines_gt.mat")

D2 <- array(as.numeric(unlist(Distribution)), dim=c(145*145,200))
DLabels <- DataLabels[[1]]
dim(DLabels) <- dim(c(145*145))

# First get rid of background pixels
row_sub <- DLabels != 0
D2 <- D2[row_sub, ]
DLabels <- DLabels[row_sub]

D2 <- cbind(as.factor(DLabels), D2)
(newColumnNames <- c("Label", paste0("WL",1:(ncol(D2)-1))))
colnames(D2) <- newColumnNames
D2 <- as.data.frame(D2)
D2$Label <- as.factor(D2$Label)

# create learning task
task1 <- TaskClassif$new(id = "CatBoostTask", D2, target = "Label")
task1

#This is loosely based on this example from the mlr3 gallery
#https://mlr3gallery.mlr-org.com/posts/2020-04-27-tuning-stacking/
svm1_lrn =  lrn("classif.svm", predict_type = "prob")
cat_lrn = lrn("classif.catboost", boosting_type='Plain', bootstrap_type='Bernoulli', 
              predict_type="prob", thread_count=4)
xgb_lrn = lrn("classif.xgboost", predict_type="prob", nthread=4)

svm1_lrn$param_set$values = list(type="C-classification", kernel="radial")

svm_cv1 = po("learner_cv", svm1_lrn, id = "svm_1", param_vals = list(resampling.folds=3))
cat_cv1 = po("learner_cv", cat_lrn, id = "cat_1", param_vals = list(resampling.folds=3))
xgb_cv1 = po("learner_cv", xgb_lrn, id = "xgb_1", param_vals = list(resampling.folds=3))

lvl1 = gunion(list(
  svm_cv1,     
  cat_cv1
  #xgb_cv1
  ))%>>% po("featureunion", id = "union1")

lvl1$plot()

ranger_final_lrn = lrn("classif.ranger", predict_type = "prob")
ensemble = lvl1 %>>% ranger_final_lrn
ensemble$plot()

#Create search space
ps_ens = ParamSet$new(
  list(
    ParamDbl$new("svm_1.cost", lower=1000, upper=5000000),
    ParamDbl$new("svm_1.gamma", lower=0.0001, upper=0.2),
    
    ParamDbl$new("cat_1.depth", 4, 10, default = 6),
    ParamDbl$new("cat_1.learning_rate", 0.001, 1, default=0.03),
    
    #ParamDbl$new("xgb_1.eta", lower = 0, upper = 1),
    #ParamDbl$new("xgb_1.nrounds", lower = 100, upper=1000),
    
    ParamDbl$new("classif.ranger.sample.fraction", lower = 0.5, upper = 1),
    ParamDbl$new("classif.ranger.num.trees", lower = 50L, upp= 1000L)
  )
)

ps_ens$trafo= function(x, param_set) {
  #x$xgb_1.nrounds = round(x$xgb_1.nrounds)
  x$cat_1.depth = round(x$cat_1.depth)
  x$classif.ranger.num.trees = round(x$classif.ranger.num.trees)
  
  x
}

ens_lrn = GraphLearner$new(ensemble)
ens_lrn$predict_type = "prob"

instance = TuningInstanceSingleCrit$new(
  task = task1,
  learner = ens_lrn,
  resampling = rsmp("holdout", ratio=0.8),
  measure = msr("classif.acc"),
  terminator = trm("evals", n_evals = 10), 
  search_space = ps_ens
)

tuner = tnr("gensa")
tuner$optimize(instance)
